src/gf-mksquashfs

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+dn=$(dirname "$0")
+# shellcheck source=src/cmdlib.sh
+. "${dn}"/cmdlib.sh
+# shellcheck source=src/libguestfish.sh
+. "${dn}"/libguestfish.sh
+
+# Usage: gf-mksquashfs <input image> <output image>
+# Example: gf-mksquashfs fedora-coreos.qcow2 fedora-coreos.squashfs
+#
+# This will generate a squashfs from the contents of the root partition.
+
+src="$1"
+dest="$2"
+
+if [[ $src == *.gz || $src == *.xz ]]; then
+    img="$(basename "$src")"
+    fatal "Cannot generate squashfs from $img; not an uncompressed image"
+fi
+
+set -x
+# Work in a tmpdir on the destination so that we don't inherit some MCS labeling
+# from the /tmp dir in the container. This also ensures that the final move is a
+# pure `rename()`.
+# See also:
+# https://github.com/coreos/coreos-assembler/issues/292
+# https://github.com/coreos/coreos-assembler/pull/394
+tmpd=$(mktemp -tdp "$(dirname "${dest}")" gf-mksquashfs.XXXXXX)
+tmp_dest=${tmpd}/image.squashfs
+
+coreos_gf_run "${src}" --ro
+
+root=$(coreos_gf findfs-label root)
+coreos_gf mount "${root}" /
```

Comment:
I know `/boot/` won't be used but wondering if we should include it for completeness? Is that was is causing this message:

```
[core@localhost ~]$ ostree admin status 
error: Unexpected state: /run/ostree-booted found and in / sysroot but not in a booted deployment
```

Reply:
We can't include the kernel and initramfs.  We could include everything else, if that's the right way to proceed.  Filed as https://github.com/ostreedev/ostree/issues/1921, and let's not block this PR for it.